### ISSUE TITLE
Relative paths of example directories not relative to doc root

### ISSUE THREAD

**Issue by florian-wagner** [sphinx-gallery/sphinx-gallery#50]
I have asked something similar in the closed issue #40, but I still think there is something wrong with the paths. Shouldn't they be relative to the documentation source (i.e. the location of conf.py) rather than relative to the location where `sphinx-build` is called?

``` bash
cd ~/tmp
git clone https://github.com/sphinx-gallery/sphinx-gallery
sphinx-build -q sphinx-gallery/doc html_output # does not work
cd sphinx-gallery/doc && sphinx-build -q . html_output # works fine
```

The output of the first `sphinx-build` call is:

``` bash
WARNING: 'default' html theme has been renamed to 'classic'. Please change your html_theme setting either to the new 'alabaster' default theme, or to 'classic' to keep using the old default.

Exception occurred:
  File "/usr/lib/python3.4/os.py", line 237, in makedirs
    mkdir(name, mode)
PermissionError: [Errno 13] Permission denied: '../../../examples'
```

**Comment by lesteve** [comment]
> Shouldn't they be relative to the documentation source (i.e. the location of conf.py) rather than relative to the location where sphinx-build is called?

I think the convention we are following is the latter, the main reason for that is that we are always building the documentation from the doc folder as far as I know. Do you see something especially wrong with that ?

**Comment by florian-wagner** [comment]
sphinx itself does allow to build the documentation everywhere and I think sphinx-gallery should not break this functionality. 

It is very useful to separate the source from compiled files including the docs. Always building the documentation in the source tree itself is especially annoying when calling the status of your version control, as a lot of untracked files are generated.

**Comment by lesteve** [comment]
Fair points. To be brutally honest, I don't see anything happening in near future on this front, just because these issues are not important enough for our use cases. Of course PRs are more than welcome!

FWIW, we are generally building documenation using the Makefile in doc rather than directly issuing sphinx commands.

**Comment by Titan-C** [comment]
This is fixed in #190
